src/dachshund/beam.rs

```diff
@@ -189,6 +191,7 @@ impl<'a, TGraph: GraphBase> Beam<'a, TGraph> {
                     scored_expansion_candidates.insert(ell);
                 }
             }
+            previous_candidates.insert(candidate.checksum.unwrap(), candidate);
```

Comment:
a comment as to what this `previous_candidates` map does (i.e., being passed as hints?) would be quite useful.